lib/Normalize_std_ast.ml

```diff
@@ -369,46 +336,49 @@ let make_mapper conf ~ignore_doc_comments ~erase_jane_syntax =
       Ast_mapper.default_mapper.class_signature m {x with pcsig_fields}
     else Ast_mapper.default_mapper.class_signature
   in
-  let label_declaration (m : Ast_mapper.mapper) ld =
-    let local_attrs, attrs =
-      extract_legacy_jane_street_local_annotations ld.pld_attributes
-    in
-    let ld =
-      { ld with
-        pld_type=
-          { ld.pld_type with
-            ptyp_attributes=
-              convert_legacy_jane_street_local_annotations
-                ~segment:Constructor_argument
-                (local_attrs @ ld.pld_type.ptyp_attributes) }
-      ; pld_attributes= attrs }
-    in
-    Ast_mapper.default_mapper.label_declaration m ld
-  in
-  let constructor_declaration (m : Ast_mapper.mapper) decl =
-    let args =
-      match decl.pcd_args with
-      | Pcstr_tuple l ->
-          Pcstr_tuple
-            (List.map
-               ~f:(fun typ ->
-                 { typ with
-                   ptyp_attributes=
-                     convert_legacy_jane_street_local_annotations
-                       ~segment:Constructor_argument typ.ptyp_attributes } )
-               l )
-      | a -> a
-    in
-    Ast_mapper.default_mapper.constructor_declaration m
-      {decl with pcd_args= args}
-  in
   let type_declaration (m : Ast_mapper.mapper) decl =
     let ptype_attributes =
       decl.ptype_attributes
-      |> List.map ~f:normalize_immediate_annot_and_attrs
+      |> List.filter_map ~f:normalize_immediate_annot_and_attrs
+      |> if erase_jane_syntax then map_attributes_no_sort m else Fn.id
     in
     Ast_mapper.default_mapper.type_declaration m {decl with ptype_attributes}
   in
+  let modes (m : Ast_mapper.mapper) ms =
+    Ast_mapper.default_mapper.modes m (if erase_jane_syntax then [] else ms)
+  in
+  let modalities (m : Ast_mapper.mapper) ms =
+    Ast_mapper.default_mapper.modalities m
+      (if erase_jane_syntax then [] else ms)
+  in
+  let value_binding (m : Ast_mapper.mapper) vb =
+    let vb =
+      match vb.pvb_expr.pexp_desc with
+      | Pexp_constraint (exp, cty, modes) when not (List.is_empty modes) ->
+          let pvb_expr =
+            match cty with
+            | None -> exp
+            | _ ->
+                {vb.pvb_expr with pexp_desc= Pexp_constraint (exp, cty, [])}
+          in
+          {vb with pvb_modes= vb.pvb_modes @ modes; pvb_expr}
+      | _ -> vb
+    in
+    Ast_mapper.default_mapper.value_binding m vb
+  in
+  let constructor_declaration (m : Ast_mapper.mapper) cd =
+    ( if erase_jane_syntax then
+        {cd with pcd_attributes= map_attributes_no_sort m cd.pcd_attributes}
```

Comment:
These steps can be stopped after jane-syntax is dead -- and probably do the wrong thing with doc comments.